src/runtime/pyscope.cs

```diff
@@ -0,0 +1,425 @@
+﻿using System;
+using System.Dynamic;
+
+namespace Python.Runtime
+{
+    public class PyScopeException : Exception
+    {
+        public PyScopeException(string message)
+            : base(message)
+        {
+
+        }
+    }
+
+    /// <summary>
+    /// Classes implement this interface must be used with GIL obtained.
+    /// </summary>
+    public interface IPyObject : IDisposable
```

Comment:
Attribute would make more sense, yes. Could allow us to test this at least at runtime.